### PR TITLE
Add documentation for Build Validation Policies for Azure Repos Git

### PR DESCRIPTION
**Pull request by jjordonm**
Currently ADO Pipelines using Azure Repos Git do not support branch policies defined within the YAML file, and instead they support them through the ADO UI configuration.

This PR adds documentation to highlight the issue and give alternatives for AzDo users on how to enforce build validation policies.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -122,6 +122,43 @@ You can use Visual Studio Code to run and debug specific tasks related to the ML
    - **Wait for Completion:** Choose `True` if you want the pipeline to wait for completion before exiting, or `False` to allow it to run asynchronously.
 5. The output and any debugging information will be displayed in the **Debug Console** or **Integrated Terminal**, depending on the task configuration.
 
+## Build Validation Policies for Azure Repos Git
+
+### Limitation in Azure DevOps Pipelines
+
+Azure DevOps Pipelines do not support PR triggers in YAML configuration for Azure Repos, while this functionality is available for GitHub. Attempting to use the `pr:` section in YAML (example shown below) will not work for Azure Repos:
```
Comment (sweanan) [inline README.md:129]: Have you tried this? https://learn.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/pr?view=azure-pipelines

Comment (kottofy) [inline README.md:129]: Can you clarify that PR triggers in YAML work in Azure Pipelines when the repo is GitHub but not when it is Azure Repos Git?

Comment (kottofy) [reply README.md:129]: Yes the note that is on that page is what this PR is for
image

Comment (jjordonm) [reply README.md]: Made this change, lmk if it has better clarification now

**Review by marshallbentley — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -122,6 +122,45 @@ You can use Visual Studio Code to run and debug specific tasks related to the ML
...
+- Navigate to your Azure Repos Git repository
+- Go to Project Settings > Repos > Branches
+- Select the branch (e.g., master or develop) where you want to enforce validation
+- Under Policies, configure a Build Validation policy to ensure that a successful build is required before a PR can be completed
+
+By doing this, you can automate build checks as part of your PR workflow, ensuring that only code passing the build can be merged into important branches.
+
+### Best Practice for YAML Defaults
+
+While branch policies handle validation, you can still configure a good default behavior in your pipeline YAML file, setting up defaults that will work in conjunction with branch policies. This allows teams to maintain a standardized pipeline configuration across branches. For more details, refer to the [Azure Pipelines YAML schema documentation](https://learn.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/?view=azure-pipelines).
```
Comment (kottofy) [inline README.md:162]: I'm not sure this section is needed?

File: README.md [hunk 3]
```diff
@@ -122,6 +122,45 @@ You can use Visual Studio Code to run and debug specific tasks related to the ML
...
+
+Follow the steps outlined in the Azure DevOps [Branch Policies documentation](https://learn.microsoft.com/en-us/azure/devops/repos/git/branch-policies?view=azure-devops&tabs=browser) to set up branch policies for build validation:
+
+- Navigate to your Azure Repos Git repository
+- Go to Project Settings > Repos > Branches
+- Select the branch (e.g., master or develop) where you want to enforce validation
+- Under Policies, configure a Build Validation policy to ensure that a successful build is required before a PR can be completed
+
+By doing this, you can automate build checks as part of your PR workflow, ensuring that only code passing the build can be merged into important branches.
+
```
Comment (kottofy) [inline README.md:159]: Can you update this to maybe use more specific steps? Something like this?
```
**To Set Up PR Triggers Using Build Validation Policies**:

**Note**: you need to have appropriate [permissions](INSERT_LINK_TO_PERMISSIONS_DOC) to create Build Validation Policies.

1. Ensure you have created the Pipeline prior to creating the Build Validation Policy.

1. Navigate to Branch Policies:
    - Go to your Azure DevOps "Project Settings".
    - Navigate to Repos > Repositories.
    - Select the Repository from the list.
    - Select the "Policies" tab.
    - Find the "Branch Policies" section and select the branch you want to set the policy for (e.g., development).

1. Add Build Validation:
    - Under "Build validation", click "+" to add a build policy.
    - Select the pipeline you want to run when a PR is created or updated.
    - Configure the policy settings, such as requiring the build to pass before completing the PR.
    - Fill in the optional box for the paths (see the [Paths](#set-path-policies) section for more information).
    - Click "Save".

For more information about Build Validation Policies, please see the [documentation](https://learn.microsoft.com/en-us/azure/devops/repos/git/branch-policies?view=azure-devops&tabs=browser#build-validation).

### Set Path Policies

Each policy contains a list of paths that specify which files or directories should trigger the pipeline when
changed. By defining these paths, we ensure that only the necessary pipelines are executed, reducing unnecessary builds and
tests, and speeding up the overall CI/CD process. For example:

- Changes to the `src/` directory trigger the build and validation pipeline.
- Changes to the `src/london_src/*` directory trigger the london_taxi pipelines and not the nyc_taxi or docker_taxi pipelines.
```

Comment (jjordonm) [reply README.md:159]: What's the permissions doc you're referring to in this?

Comment (kottofy) [reply README.md:159]: How is this one? https://learn.microsoft.com/en-us/azure/devops/repos/git/branch-policies?view=azure-devops&tabs=browser#prerequisites

Comment (jjordonm) [reply README.md]: Made this change

File: README.md (hunk 2 above)
Comment (jjordonm) [reply README.md]: Deleted this section

**Review by kottofy — APPROVED** [review]

**Pull request merged**